code/game/gamemodes/colonialmarines/colonialmarines.dm

```diff
@@ -358,6 +359,84 @@
 		var/rendered_announce_text = replacetext(SSmapping.configs[GROUND_MAP].announce_text, "###SHIPNAME###", MAIN_SHIP_NAME)
 		marine_announcement(rendered_announce_text, "[MAIN_SHIP_NAME]")
 
+/datum/game_mode/proc/ares_command_check()
+	var/role_in_charge
+	var/mob/living/carbon/human/person_in_charge
+	var/update_id = FALSE
+	var/doesnotneedheadset = FALSE
+	var/announce_addendum
+
+
+	var/datum/job/command/auxiliary_officer/aso_job = GLOB.RoleAuthority.roles_for_mode[JOB_AUXILIARY_OFFICER]
+	var/datum/job/command/warrant/cmp_job = GLOB.RoleAuthority.roles_for_mode[JOB_CHIEF_POLICE]
+	var/datum/job/civilian/professor/cmo_job = GLOB.RoleAuthority.roles_for_mode[JOB_CMO]
+	var/datum/job/command/pilot/dropship_pilot/dp_job = GLOB.RoleAuthority.roles_for_mode[JOB_DROPSHIP_PILOT]
+	var/datum/job/command/pilot/cas_pilot/gs_job = GLOB.RoleAuthority.roles_for_mode[JOB_CAS_PILOT]
+	var/datum/job/logistics/engineering/ce_job = GLOB.RoleAuthority.roles_for_mode[JOB_CHIEF_ENGINEER]
+	var/datum/squad/intel_squad = GLOB.RoleAuthority.squads_by_type[/datum/squad/marine/intel]
+	var/list/intel_officers = intel_squad.marines_list
```

Comment:
This is horrible. make a proc in /datum/job called "get_active_job_player" (doesnt matter) that returns nothing, then override that in jobs where you need to get the active jobs, they will return the active player if any. then a single for loop here. someone in the future will say thank you aswell.